### DISCUSSION TITLE
Which connect client/receiver uses auth type access-token ?

### DISCUSSION THREAD

**Discussion by kingosticks** [TimotheeGerber/spotify-connect#2] [Q&A] [answered]
Librespot doesn't yet have any support for discovery-based login using the access-token style auth type, presumably because the official clients don't use it. So I was wondering what connect receiver was that tested against? And similarly, what client prefers that method?

**Comment by TimotheeGerber** [comment, marked as answer]
Early in the connection process, receivers are sending information about themselves. The official Spotify client then send an authentication request depending on the receiver information. In particular, the presence and value of `tokenType` in the `getInfo` response change the authentication request content. You can read more information I discovered in [this comment of issue #1](https://github.com/TimotheeGerber/spotify-connect/issues/1#issuecomment-1083622041).

I don't know if `librespot` should support `accesstoken` authentication type. As long as the official Sportify client is backward compatible and support the legacy methods, it should be fine. 

---

I did not do the tests myself, but @hillstub needed the `accesstoken` method to authenticate its **Denon AVR-X1600H**. I don't have any other news about successful connection with other officially supported devices.

**Reply by kingosticks** [reply]
Perfect thanks. It's easy to add that support in librespot and there's a PR for it now. Just wanted some background.

P.s. your tool is great. Maybe we should merge some of it into librespot e.g. blob encryption.

**Reply by TimotheeGerber** [reply]
Thank you! I am glad to see that this project is useful to others than me.

Feel free to integrate whatever part of this project into `librespot`. If you need any help, let me know and I will try my best to find time.
